ParallelMatrixMultiplication/ParallelMatrixMultiplication/Matrix.cs

```diff
@@ -0,0 +1,140 @@
+        /// </summary>
+        /// <param name="rows">Number of lines.</param>
+        /// <param name="columns">Number of columns.</param>
+        /// <returns>Matrix with random values.</returns>
+        public static Matrix GenerateRandomMatrix(int rows, int columns)
+        {
+            Random random = new Random();
+            Matrix matrix = new Matrix(rows, columns);
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < columns; ++j)
+                {
+                    matrix[i, j] = random.Next(1, 74);
+                }
+            }
+
+            return matrix;
+        }
+
+        /// <summary>
+        /// A method for loading a matrix from a file.
+        /// </summary>
+        /// <param name="filePath">The path to the matrix file.</param>
+        /// <returns>The matrix from the file.</returns>
+        /// <exception cref="FileNotFoundException">It is thrown if the file is not found.</exception>
+        /// <exception cref="InvalidDataException">It is thrown if the file is empty or contains incorrect data.</exception>
+        /// <exception cref="FormatException">It is thrown if the numbers in the file have an incorrect format.</exception>
+        public static Matrix LoadFromFile(string filePath)
+        {
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"Файл {filePath} не найден");
+            }
+
+            string[] lines = File.ReadAllLines(filePath);
+
+            if (lines.Length == 0)
+            {
+                throw new InvalidDataException("Файл пуст");
+            }
+
+            int rows = lines.Length;
+
+            string[] firstRowValues = lines[0].Split(' ', StringSplitOptions.RemoveEmptyEntries);
+            int columnsSize = firstRowValues.Length;
+
+            Matrix matrix = new Matrix(rows, columnsSize);
+
+            for (int i = 0; i < rows; ++i)
+            {
+                string[] values = lines[i].Split(' ', StringSplitOptions.RemoveEmptyEntries);
+
+                if (values.Length != columnsSize)
+                {
+                    throw new InvalidDataException("Строка имеет некорректное значение стобцов");
+                }
+
+                for (int j = 0; j < columnsSize; ++j)
+                {
+                    if (!int.TryParse(values[j], out int parsedValue))
+                    {
+                        throw new FormatException("Неверный формат числа в строке");
+                    }
+
+                    matrix[i, j] = parsedValue;
+                }
+            }
+
+            return matrix;
+        }
+
+        /// <summary>
+        /// A method for saving a matrix to a file.
+        /// </summary>
+        /// <param name="filePath">The path to save the file.</param>
+        public void SaveToFile(string filePath)
+        {
+            using (StreamWriter writer = new StreamWriter(filePath))
```

Comment:
```suggestion
            using StreamWriter writer = new StreamWriter(filePath);
```
 для таких случаев подойдет using declaration: тоже, чтобы не увеличивать уровень вложенности.
 довольно удобно, а Java такое сделать тоже не даст